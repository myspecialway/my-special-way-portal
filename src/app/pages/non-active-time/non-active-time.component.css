/**
 * Flex-based table structure
 */
mat-table {
  display: block;
  direction: rtl;
  font-family: 'Rubik', 'Helvetica', 'Arial', sans-serif;
}

mat-paginator {
  direction: rtl;
}

mat-header-row {
  background-color: #f2f2f2;
  min-height: 48px;
}

mat-row,
mat-footer-row {
  min-height: 50px;
}

mat-row,
mat-header-row,
mat-footer-row {
  display: flex;
  border-width: 0;
  border-bottom-width: 1px;
  border-style: solid;
  align-items: center;
  box-sizing: border-box;
}
mat-row::after,
mat-header-row::after,
mat-footer-row::after {
  display: inline-block;
  min-height: inherit;
  content: '';
}

mat-cell:first-child,
mat-header-cell:first-child,
mat-footer-cell:first-child {
  padding-left: 24px;
}
[dir='rtl'] mat-cell:first-child,
[dir='rtl'] mat-header-cell:first-child,
[dir='rtl'] mat-footer-cell:first-child {
  padding-left: 0;
  padding-right: 24px;
}

mat-cell:last-child,
mat-header-cell:last-child,
mat-footer-cell:last-child {
  padding-right: 24px;
}
[dir='rtl'] mat-cell:last-child,
[dir='rtl'] mat-header-cell:last-child,
[dir='rtl'] mat-footer-cell:last-child {
  padding-right: 0;
  padding-left: 24px;
}

mat-cell,
mat-header-cell,
mat-footer-cell {
  flex: 1;
  display: flex;
  align-items: center;
  overflow: hidden;
  word-wrap: break-word;
  min-height: inherit;
}

/**
 * Native HTML table structure
 */
table.mat-table {
  border-spacing: 0;
}

tr.mat-header-row {
  height: 48px;
}

tr.mat-row,
tr.mat-footer-row {
  height: 50px;
}

th.mat-header-cell {
  text-align: left;
}

th.mat-header-cell,
td.mat-cell,
td.mat-footer-cell {
  padding: 0;
  border-bottom: 1px solid;
}

th.mat-header-cell:first-child,
td.mat-cell:first-child,
td.mat-footer-cell:first-child {
  padding-left: 24px;
}

th.mat-header-cell:last-child,
td.mat-cell:last-child,
td.mat-footer-cell:last-child {
  padding-right: 24px;
}

.msw-table-tools-header {
  align-items: center;
  display: flex;
  justify-content: space-between;
}

mat-row:hover {
  background-color: #e1ccf5;
  cursor: pointer;
}
mat-row:hover mat-cell {
  color: #8c37cd;
}
mat-row:hover .delete-row-button {
  display: block;
}

.delete-row-button {
  display: none;
}

.mat-raised-button.mat-primary {
  background-color: #6a00cf;
  border-radius: 3px;
}

.mat-tab-group.mat-primary .mat-ink-bar,
.mat-tab-nav-bar.mat-primary .mat-ink-bar {
  background-color: #ff0;
}

.header-table-count {
  width: 164px;
  height: 38px;
  font-size: 32px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: 0.6px;
  text-align: right;
  color: #222222;
}

.header-table-count-number {
  font-size: 18px;
}

.header-table-count-label {
  padding-left: 10px;
}

.header-menu-container {
  padding-bottom: 29px;
}

.msw-table-container .mat-cell:last-child,
.msw-table-container .mat-header-cell:last-child {
  flex-grow: 0;
  flex-basis: 100px;
}

.msw-table-container .mat-header-cell {
  font-weight: bold;
}

.no-records-msg {
  padding: 100px;
  text-align: center;
}

.filter-toggle-button {
  font-size: large;
  padding-top: 0.5rem;
}

.filter {
  max-width: 70%;
}

.material-icons.md-18 {
  font-size: 18px;
}

.material-icons.md-24 {
  font-size: 24px;
}

.material-icons.md-36 {
  font-size: 36px;
}

.material-icons.md-48 {
  font-size: 48px;
}

.header-table-count-label {
  font-size: 21px;
}

/*# sourceMappingURL=non-active-time.component.css.map */
