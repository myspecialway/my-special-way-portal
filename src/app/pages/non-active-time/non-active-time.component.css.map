{"version":3,"sourceRoot":"","sources":["../../../assets/scss/msw-table.scss","non-active-time.component.scss"],"names":[],"mappings":"AAIA;AAAA;AAAA;AAGA;EACE;EACA;EACA;;;AAGF;EACE;;;AAGF;EACE;EACA,YAnBsB;;;AAsBxB;AAAA;EAEE,YAvBe;;;AA0BjB;AAAA;AAAA;EAGE;EAGA;EACA;EACA;EACA;EACA;;AAKA;AAAA;AAAA;EACE;EACA;EACA;;;AAIJ;AAAA;AAAA;EAGE,cAlD2B;;AAoD3B;AAAA;AAAA;EACE;EACA,eAtDyB;;;AA0D7B;AAAA;AAAA;EAGE,eA7D2B;;AA+D3B;AAAA;AAAA;EACE;EACA,cAjEyB;;;AAqE7B;AAAA;AAAA;EAGE;EACA;EACA;EACA;EACA;EACA;;;AAGF;AAAA;AAAA;AAGA;EACE;;;AAGF;EACE,QA1FsB;;;AA6FxB;AAAA;EAEE,QA9Fe;;;AAiGjB;EACE;;;AAGF;AAAA;AAAA;EAGE;EACA;;;AAGF;AAAA;AAAA;EAGE,cA9G2B;;;AAiH7B;AAAA;AAAA;EAGE,eApH2B;;;AAuH7B;EACE;EACA;EACA;;;AAGF;EACE;EACA;;AACA;EACE;;AAEF;EACE;;;AAIJ;EACE;;;AAGF;EACE;EACA;;;AAGF;AAAA;EAEE;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;AAAA;EAEE;EACA;;;AAEF;EACE;;;AAGF;EACE;EACA;;;AAGF;EACE;EACA;;;AAGF;EACE;;;AAGF;EACE;;;AAEF;EACE;;;AAEF;EACE;;;AAEF;EACE;;;AC/MF;EACE","file":"non-active-time.component.css"}