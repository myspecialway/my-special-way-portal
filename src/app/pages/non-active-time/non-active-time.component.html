<div data-test-id="non-active-time-page" class="main-content msw-non-active-time-container">
  <div class="container-fluid">
    <div class="header-menu-container msw-table-tools-header">
      <div class="header-table-count">
        <div>
          <span class="header-table-count-label">זמני אי פעילות</span>
          <span class="header-table-count-number">({{dataSource.data.length}})</span>
        </div>
      </div>
      <div class="example-header msw-table-tools-header">
        <button data-test-id="add-new-non-active-time-button" mat-raised-button color="primary" routerLink="/student/_new_">
          <i class="material-icons">
            add
          </i>
          הוסף זמן אי פעילות
        </button>
      </div>
    </div>
    <div class="row msw-table-container">
      <div class="col-lg-12 col-md-12 col-sm-12">
        <mat-table #table [dataSource]="dataSource" class="msw-table" matSort matSortActive="title" matSortDisableClear
                   matSortDirection="asc" >

          <!-- Name Column -->
          <ng-container matColumnDef="title" >
            <mat-header-cell *matHeaderCellDef mat-sort-header>אירוע</mat-header-cell>
            <mat-cell *matCellDef="let row" (click)="editNewNonActiveTime(row)">
              <div>{{row.title}}</div>
            </mat-cell>
          </ng-container>

          <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
          <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
        </mat-table>
      </div>
    </div>
  </div>
</div>
