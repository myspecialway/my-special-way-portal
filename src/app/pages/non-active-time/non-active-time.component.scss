@import '../../../assets/scss/msw-table';
