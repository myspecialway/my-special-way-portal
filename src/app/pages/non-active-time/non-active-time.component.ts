import { Component, OnInit, ViewChild } from '@angular/core';
import { MatDialog, MatSort, MatTableDataSource } from '@angular/material';
import { SubscriptionCleaner } from '../../decorators/SubscriptionCleaner.decorator';
import { Subscription } from 'rxjs';
import { NonActiveTimeService } from '../../services/non-active-time/non-active-time.graphql.service';
import { NonActiveTime } from '../../models/non-active-time.model';
import { first } from 'rxjs/operators';
import { DeleteNonActiveTimeDialogueComponent } from './delete/delete-non-active-time-dialogue.component';

@Component({
  selector: 'app-non-active-time',
  templateUrl: './non-active-time.component.html',
  styleUrls: ['./non-active-time.component.scss'],
})
export class NonActiveTimeComponent implements OnInit {
  displayedColumns = ['title', 'classes', 'dates', 'hours', 'deleteNonActiveTime'];
  dataSource = new MatTableDataSource<NonActiveTime>();

  @ViewChild(MatSort)
  sort: MatSort;

  @SubscriptionCleaner()
  subCollector: Subscription;

  constructor(private nonActiveTimeService: NonActiveTimeService, public dialog: MatDialog) {}

  async ngOnInit() {
    try {
      this.subCollector.add(
        this.nonActiveTimeService.getAllNonActiveTimes().subscribe((nonActiveTimes) => {
          if (nonActiveTimes !== null) {
            this.dataSource.data = [...nonActiveTimes];
          } else {
            this.dataSource.data = [];
          }
        }),
      );
      this.dataSource.sort = this.sort;
    } catch (error) {
      console.error('Error getting non active times');
      throw error;
    }
  }

  public async deleteNonActiveTime(nonActiveTime: NonActiveTime) {
    const dialogRef = this.dialog.open(DeleteNonActiveTimeDialogueComponent, {
      data: { _id: nonActiveTime._id, title: nonActiveTime.title },
    });
    this.subCollector.add(
      dialogRef
        .afterClosed()
        .pipe(first())
        .subscribe(async (result) => {
          if (result === true) {
            this.nonActiveTimeService.delete(nonActiveTime._id);
          }
        }),
    );
  }

  getClassesDisplayData(row: NonActiveTime): string {
    if (row.isAllClassesEvent) return 'כל הכיתות';
    if (!row.classes) throw new Error('Non active time is marked for specific classes, but no classes were specified');
    if (row.classes.length === 1) return row.classes[0].name;
    return row.classes[0].name + ' + ' + (row.classes.length - 1);
  }

  getDatesDisplayData(row: NonActiveTime): string {
    const startDateTime = new Date(row.startDateTime);
    const options = { year: 'numeric', month: 'long', day: 'numeric' };
    let result: string = startDateTime.toLocaleDateString('he-IL', options);
    if (row.startDateTime !== row.endDateTime) {
      const endDateTime = new Date(row.startDateTime);
      result += ' עד ' + endDateTime.toLocaleDateString('he-IL', options);
    }

    return result;
  }

  getHoursDisplayData(row: NonActiveTime): string {
    if (row.isAllDayEvent) return 'כל היום';
    const startDateTime = new Date(row.startDateTime);
    const endDateTime = new Date(row.endDateTime);
    const options = { hour: 'numeric', minute: 'numeric' };
    return (
      startDateTime.toLocaleTimeString('he-IL', options) + ' עד ' + endDateTime.toLocaleTimeString('he-IL', options)
    );
  }
}
