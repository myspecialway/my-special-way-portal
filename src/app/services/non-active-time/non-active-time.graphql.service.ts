import { Injectable } from '@angular/core';
import gql from 'graphql-tag';
import { Apollo } from 'apollo-angular';
import { Observable, of as observableOf, Subscription } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import { NonActiveTime } from '../../models/non-active-time.model';
import { FetchResult } from 'apollo-link';
import { SubscriptionCleaner } from '../../decorators/SubscriptionCleaner.decorator';

const GET_ALL_NON_ACTIVE_TIMES_QUERY = gql`
  {
    nonActiveTimes {
      _id
      title
      isAllDayEvent
      startDateTime
      endDateTime
      isAllClassesEvent
      classes {
        _id
        name
      }
    }
  }
`;

const DELETE_NON_ACTIVE_TIME_QUERY = (id: string) => gql`
mutation {
  deleteNonActiveTime(
    id: "${id}"
  )
}`;

const CREATE_NON_ACTIVE_TIME_QUERY = (
  title: string,
  isAllDayEvent: boolean,
  startDateTime: string,
  endDateTime: string,
  isAllClassesEvent: boolean,
  classesIds: string[] | undefined,
) => gql`
  mutation {
    createNonActiveTime(nonActiveTime: {
      title: "${title}"
      isAllDayEvent: ${isAllDayEvent}
      startDateTime: "${startDateTime}"
      endDateTime: "${endDateTime}"
      isAllClassesEvent: ${isAllClassesEvent}
      classesIds: ${classesIds ? '[' + classesIds.map((classesId) => `"${classesId}"`) + ']' : undefined}
    }){ _id }
  }`;

const UPDATE_NON_ACTIVE_TIME_QUERY = (
  id: string,
  title: string,
  isAllDayEvent: boolean,
  startDateTime: string,
  endDateTime: string,
  isAllClassesEvent: boolean,
  classesIds: string[] | undefined,
) => gql`
 mutation {
  updateNonActiveTime(id: "${id}", nonActiveTime: {
    title: "${title}"
    isAllDayEvent: ${isAllDayEvent}
    startDateTime: "${startDateTime}"
    endDateTime: "${endDateTime}"
    isAllClassesEvent: ${isAllClassesEvent}
    classesIds: ${classesIds ? '[' + classesIds.map((classesId) => `"${classesId}"`) + ']' : undefined}
  }){ _id }
}`;

@Injectable()
export class NonActiveTimeService {
  @SubscriptionCleaner()
  subCollector: Subscription;

  constructor(private apollo: Apollo) {}

  public getAllNonActiveTimes(): Observable<NonActiveTime[]> {
    return this.apollo
      .watchQuery<{ nonActiveTimes: NonActiveTime[] }>({
        query: GET_ALL_NON_ACTIVE_TIMES_QUERY,
      })
      .valueChanges.pipe(
        map((res) => {
          return res.data.nonActiveTimes;
        }),
        catchError((err: TypeError) => {
          console.warn('user.component::ngInInit:: empty stream received');
          return observableOf([]);
        }),
      );
  }

  public create(
    title: string,
    isAllDayEvent: boolean,
    startDateTime: string,
    endDateTime: string,
    isAllClassesEvent: boolean,
    classesIds: string[] | undefined,
  ): any {
    return this.apollo
      .mutate({
        mutation: CREATE_NON_ACTIVE_TIME_QUERY(
          title,
          isAllDayEvent,
          startDateTime,
          endDateTime,
          isAllClassesEvent,
          classesIds,
        ),
        refetchQueries: [
          {
            query: GET_ALL_NON_ACTIVE_TIMES_QUERY,
          },
        ],
        awaitRefetchQueries: true,
      })
      .toPromise();
  }

  public update(
    id: string,
    title: string,
    isAllDayEvent: boolean,
    startDateTime: string,
    endDateTime: string,
    isAllClassesEvent: boolean,
    classesIds: string[] | undefined,
  ): any {
    return this.apollo
      .mutate({
        mutation: UPDATE_NON_ACTIVE_TIME_QUERY(
          id,
          title,
          isAllDayEvent,
          startDateTime,
          endDateTime,
          isAllClassesEvent,
          classesIds,
        ),
        refetchQueries: [
          {
            query: GET_ALL_NON_ACTIVE_TIMES_QUERY,
          },
        ],
        awaitRefetchQueries: true,
      })
      .toPromise();
  }

  public async delete(nonActiveTimeId: string): Promise<FetchResult<Record<string, any>>> {
    return this.apollo
      .mutate({
        mutation: DELETE_NON_ACTIVE_TIME_QUERY(nonActiveTimeId),
        refetchQueries: [
          {
            query: GET_ALL_NON_ACTIVE_TIMES_QUERY,
          },
        ],
        awaitRefetchQueries: true,
      })
      .toPromise();
  }
}
