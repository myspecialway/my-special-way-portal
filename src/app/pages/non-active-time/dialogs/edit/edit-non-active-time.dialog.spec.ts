import {
  MatInputModule,
  MatDialogModule,
  MatSelectModule,
  MatCheckboxModule,
  MatDividerModule,
  MatAutocompleteModule,
  MatRadioModule,
  MatDatepickerModule,
  MatNativeDateModule,
  MatDialogRef,
  MAT_DIALOG_DATA,
} from '@angular/material';
import { TestBed } from '@angular/core/testing';
import { EditNonActiveTimeDialogComponent } from './edit-non-active-time.dialog';
import { NonActiveTime } from '../../../../models/non-active-time.model';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { of } from 'rxjs';
import { classTestData } from '../../../../../mocks/assets/classes.mock';
import { ClassService } from '../../../class/services/class.graphql.service';

describe('EditNonActiveTimeDialogComponent', () => {
  let dialogData: NonActiveTime;
  let classServiceMock: Partial<ClassService>;
  let matDialogRefMock: Partial<MatDialogRef<EditNonActiveTimeDialogComponent>>;

  beforeEach(async () => {
    dialogData = {
      _id: '123',
      title: '',
      isAllDayEvent: true,
      startDateTime: new Date().toUTCString(),
      endDateTime: new Date().toUTCString(),
      isAllClassesEvent: true,
      classes: undefined,
    };

    classServiceMock = {
      getAllClasses: jest.fn().mockImplementation(() => of(classTestData.classes)),
    };

    matDialogRefMock = {
      close: jest.fn(),
    };

    TestBed.configureTestingModule({
      imports: [
        FormsModule,
        ReactiveFormsModule,
        MatInputModule,
        MatDialogModule,
        MatSelectModule,
        MatCheckboxModule,
        MatDividerModule,
        MatAutocompleteModule,
        MatRadioModule,
        MatDatepickerModule,
        MatNativeDateModule,
      ],
      declarations: [EditNonActiveTimeDialogComponent],
      providers: [
        { provide: MatDialogRef, useValue: matDialogRefMock },
        { provide: ClassService, useValue: classServiceMock },
        { provide: MAT_DIALOG_DATA, useValue: dialogData },
      ],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should render component as described in snapshot', () => {
    const fixture = TestBed.createComponent(EditNonActiveTimeDialogComponent);
    expect(fixture).toMatchSnapshot();
  });

  it('should output undefined if dialog was canceled', () => {
    const fixture = TestBed.createComponent(EditNonActiveTimeDialogComponent);
    fixture.componentInstance.onNoClick();
    expect(matDialogRefMock.close).toHaveBeenCalledWith();
  });

  it('should output correct NonActiveTime object if dialog was submited', () => {
    const startDateTime = new Date('Tue, 11 Dec 2018 00:00:00 GMT');
    const endDateTime = new Date('Tue, 11 Dec 2018 01:00:00 GMT');
    const fixture = TestBed.createComponent(EditNonActiveTimeDialogComponent);
    fixture.componentInstance.submitForm(
      fixture.componentInstance.formBuilder.group({
        title: ['a'],
        isAllDayEvent: [false],
        startDateTime: [startDateTime],
        endDateTime: [endDateTime],
        startHour: ['09:00'],
        endHour: ['11:00'],
        isAllClassesEvent: [false],
        classes: [[{ _id: '5b10fb1ff8022f6011f30f48', name: 'טיטאן' }]],
      }),
    );
    expect(matDialogRefMock.close).toHaveBeenCalledWith({
      _id: dialogData._id,
      title: 'a',
      isAllDayEvent: false,
      startDateTime: 'Tue, 11 Dec 2018 07:00:00 GMT',
      endDateTime: 'Tue, 11 Dec 2018 09:00:00 GMT',
      isAllClassesEvent: false,
      classes: [{ _id: '5b10fb1ff8022f6011f30f48', name: 'טיטאן' }],
    });
  });
});
