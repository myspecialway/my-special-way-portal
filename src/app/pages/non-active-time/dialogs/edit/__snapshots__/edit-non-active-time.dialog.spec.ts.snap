// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNonActiveTimeDialogComponent should render component as described in snapshot 1`] = `
<app-edit-non-active-time
  addButtonTitle={[Function String]}
  addDialogTitle={[Function String]}
  classService={[Function Object]}
  data={[Function Object]}
  dialogRef={[Function Object]}
  formBuilder={[Function FormBuilder]}
  hours={[Function Array]}
  updateButtonTitle={[Function String]}
  updateDialogTitle={[Function String]}
>
  <form
    novalidate=""
  >
    <div
      class="container"
    >
      <h3
        class="mat-dialog-title"
        mat-dialog-title=""
      >
        
      </h3>
      <div
        class="mat-dialog-content"
        mat-dialog-content=""
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <mat-form-field
              class="w-100 mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      formcontrolname="title"
                      matinput=""
                      name="title"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col w-25"
          >
            <mat-form-field
              class="app-input-no-underline app-form-field-no-padding-bottom mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      matinput=""
                      style="display: none"
                    />
                    <mat-radio-group
                      class="mat-radio-group"
                      formcontrolname="isAllClassesEvent"
                      role="radiogroup"
                    >
                      <mat-radio-button
                        class="mat-radio-button mat-accent"
                        color="primary"
                      >
                        <label
                          class="mat-radio-label"
                        >
                          <div
                            class="mat-radio-container"
                          >
                            <div
                              class="mat-radio-outer-circle"
                            />
                            <div
                              class="mat-radio-inner-circle"
                            />
                            <div
                              class="mat-radio-ripple mat-ripple"
                              mat-ripple=""
                            />
                          </div>
                          <input
                            class="mat-radio-input cdk-visually-hidden"
                            type="radio"
                          />
                          <div
                            class="mat-radio-label-content"
                          >
                            <span
                              style="display:none"
                            >
                               
                            </span>
                             בחר כיתה 
                          </div>
                        </label>
                      </mat-radio-button>
                    </mat-radio-group>
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
          <div
            class="col w-50"
          >
            <mat-form-field
              class="app-form-field-no-padding-bottom mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <mat-select
                      class="mat-select"
                      formcontrolname="classes"
                      multiple=""
                      role="listbox"
                    >
                      <div
                        aria-hidden="true"
                        cdk-overlay-origin=""
                        class="mat-select-trigger"
                      >
                        <div
                          class="mat-select-value"
                        >
                          
                          
                        </div>
                        <div
                          class="mat-select-arrow-wrapper"
                        >
                          <div
                            class="mat-select-arrow"
                          />
                        </div>
                      </div>
                      
                    </mat-select>
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
            
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col w-25"
          >
            <mat-form-field
              class="app-input-no-underline app-form-field-no-border app-form-field-no-padding-bottom mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      matinput=""
                      style="display: none"
                    />
                    <mat-radio-group
                      class="mat-radio-group"
                      formcontrolname="isAllClassesEvent"
                      role="radiogroup"
                    >
                      <mat-radio-button
                        class="mat-radio-button mat-accent"
                        color="primary"
                      >
                        <label
                          class="mat-radio-label"
                        >
                          <div
                            class="mat-radio-container"
                          >
                            <div
                              class="mat-radio-outer-circle"
                            />
                            <div
                              class="mat-radio-inner-circle"
                            />
                            <div
                              class="mat-radio-ripple mat-ripple"
                              mat-ripple=""
                            />
                          </div>
                          <input
                            class="mat-radio-input cdk-visually-hidden"
                            type="radio"
                          />
                          <div
                            class="mat-radio-label-content"
                          >
                            <span
                              style="display:none"
                            >
                               
                            </span>
                             כל הכיתות 
                          </div>
                        </label>
                      </mat-radio-button>
                    </mat-radio-group>
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
        </div>
        <mat-divider
          class="mat-divider"
          role="separator"
        />
        <div
          class="row"
        >
          <div
            class="col date-input-field"
          >
            <mat-form-field
              class="date mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      formcontrolname="startDateTime"
                      matinput=""
                      readonly=""
                    />
                    <mat-datepicker
                      touchui=""
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
          <div
            class="col label-between-fields"
          >
            <span>
              עד
            </span>
          </div>
          <div
            class="col date-input-field"
          >
            <mat-form-field
              class="date mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      formcontrolname="endDateTime"
                      matinput=""
                      readonly=""
                    />
                    <mat-datepicker
                      touchui=""
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
        </div>
        <mat-divider
          class="mat-divider"
          role="separator"
        />
        <div
          class="row"
        >
          <div
            class="col w-25"
          >
            <mat-form-field
              class="app-input-no-underline app-form-field-no-padding-bottom mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      matinput=""
                      style="display: none"
                    />
                    <mat-radio-group
                      class="mat-radio-group"
                      formcontrolname="isAllDayEvent"
                      role="radiogroup"
                    >
                      <mat-radio-button
                        class="mat-radio-button mat-accent"
                        color="primary"
                      >
                        <label
                          class="mat-radio-label"
                        >
                          <div
                            class="mat-radio-container"
                          >
                            <div
                              class="mat-radio-outer-circle"
                            />
                            <div
                              class="mat-radio-inner-circle"
                            />
                            <div
                              class="mat-radio-ripple mat-ripple"
                              mat-ripple=""
                            />
                          </div>
                          <input
                            class="mat-radio-input cdk-visually-hidden"
                            type="radio"
                          />
                          <div
                            class="mat-radio-label-content"
                          >
                            <span
                              style="display:none"
                            >
                               
                            </span>
                             שעות 
                          </div>
                        </label>
                      </mat-radio-button>
                    </mat-radio-group>
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
          <div
            class="col w-50"
          >
            <div
              class="row"
            >
              <div
                class="col date-input-field"
              >
                <mat-form-field
                  class="hour app-form-field-no-padding-bottom mat-form-field mat-primary"
                >
                  <div
                    class="mat-form-field-wrapper"
                  >
                    <div
                      class="mat-form-field-flex"
                    >
                      
                      
                      <div
                        class="mat-form-field-infix"
                      >
                        <input
                          class="text-center mat-input-element mat-form-field-autofill-control"
                          formcontrolname="startHour"
                          matinput=""
                          type="text"
                        />
                        <span
                          class="mat-form-field-label-wrapper"
                        >
                          
                        </span>
                      </div>
                      
                    </div>
                    
                    <div
                      class="mat-form-field-subscript-wrapper"
                    >
                      
                      
                    </div>
                  </div>
                </mat-form-field>
                <mat-autocomplete
                  class="mat-autocomplete"
                >
                  
                </mat-autocomplete>
                
              </div>
              <div
                class="col label-between-fields"
              >
                <span>
                  עד
                </span>
              </div>
              <div
                class="col date-input-field"
              >
                <mat-form-field
                  class="hour app-form-field-no-padding-bottom mat-form-field mat-primary"
                >
                  <div
                    class="mat-form-field-wrapper"
                  >
                    <div
                      class="mat-form-field-flex"
                    >
                      
                      
                      <div
                        class="mat-form-field-infix"
                      >
                        <input
                          class="text-center mat-input-element mat-form-field-autofill-control"
                          formcontrolname="endHour"
                          matinput=""
                          type="text"
                        />
                        <span
                          class="mat-form-field-label-wrapper"
                        >
                          
                        </span>
                      </div>
                      
                    </div>
                    
                    <div
                      class="mat-form-field-subscript-wrapper"
                    >
                      
                      
                    </div>
                  </div>
                </mat-form-field>
                <mat-autocomplete
                  class="mat-autocomplete"
                >
                  
                </mat-autocomplete>
                
              </div>
              
            </div>
            
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col w-25"
          >
            <mat-form-field
              class="app-input-no-underline app-form-field-no-border app-form-field-no-padding-bottom mat-form-field mat-primary"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      class="mat-input-element mat-form-field-autofill-control"
                      matinput=""
                      style="display: none"
                    />
                    <mat-radio-group
                      class="mat-radio-group"
                      formcontrolname="isAllDayEvent"
                      role="radiogroup"
                    >
                      <mat-radio-button
                        class="mat-radio-button mat-accent"
                        color="primary"
                      >
                        <label
                          class="mat-radio-label"
                        >
                          <div
                            class="mat-radio-container"
                          >
                            <div
                              class="mat-radio-outer-circle"
                            />
                            <div
                              class="mat-radio-inner-circle"
                            />
                            <div
                              class="mat-radio-ripple mat-ripple"
                              mat-ripple=""
                            />
                          </div>
                          <input
                            class="mat-radio-input cdk-visually-hidden"
                            type="radio"
                          />
                          <div
                            class="mat-radio-label-content"
                          >
                            <span
                              style="display:none"
                            >
                               
                            </span>
                             כל היום 
                          </div>
                        </label>
                      </mat-radio-button>
                    </mat-radio-group>
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-subscript-wrapper"
                >
                  
                  
                </div>
              </div>
            </mat-form-field>
          </div>
        </div>
      </div>
      <div
        class="mat-dialog-actions"
        mat-dialog-actions=""
      >
        <button
          class="msw-button-cancel"
          data-test-id="cancel-button"
          mat-button=""
          tabindex="-1"
          type="button"
        >
           ביטול 
        </button>
        <button
          class="msw-button-submit"
          data-test-id="submit-button"
          mat-button=""
          type="submit"
        >
           
        </button>
      </div>
    </div>
  </form>
</app-edit-non-active-time>
`;
